hazelcast/src/main/java/com/hazelcast/sql/impl/SqlInternalService.java

```diff
@@ -209,4 +210,33 @@ public QueryOperationHandlerImpl getOperationHandler() {
     public QueryClientStateRegistry getClientStateRegistry() {
         return clientStateRegistry;
     }
+
+    private void prepareParameters(Plan plan, List<Object> params) {
+        assert params != null;
+        QueryParameterMetadata parameterMetadata = plan.getParameterMetadata();
+
+        int parameterCount = parameterMetadata.getParameterCount();
+        if (parameterCount != params.size()) {
+            throw QueryException.error(SqlErrorCode.DATA_EXCEPTION,
+                    "Unexpected parameter count: expected " + parameterCount + ", got " + params.size());
+        }
+
+        for (int i = 0; i < params.size(); ++i) {
+            Object value = params.get(i);
+            if (value == null) {
+                continue;
+            }
+
+            Converter valueConverter = Converters.getConverter(value.getClass());
+            Converter typeConverter = parameterMetadata.getParameterType(i).getConverter();
+            try {
+                value = typeConverter.convertToSelf(valueConverter, value);
+            } catch (QueryException e) {
```

Comment:
`catch (QueryException)` misses other types of exceptions. It seems we need `catch (Exception)` here

Reply:
Other types of exceptions are not related to conversion, users should not deal with them under the normal conditions.

Reply:
Not really. Consider `DoubleConverter`. In the current branch, it cannot process conversion to `DECIMAL` properly in case of NaN and infinity. The result is an arithmetic exception with the error message `Infinite or NaN`. This is precisely what the user gets in the end: an `SqlException` with errorCode=-1 (GENERIC) and the message `Infinite or NaN`. There is no way for him to understand what is the real problem.

We cannot guarantee that we handle all exceptions in our converters. We may get all sorts of problems there, but users should get proper hint on where the exception happened - when attempting to convert this specific parameter to this specific type.

Reply:
fixd in d5fae6a332a981393bd69a8defaca5b42c46ce90